In [1]:
import pandas as pd

In [2]:
airbnb = pd.read_csv('airbnb_data.csv')

In [3]:
%%sql
SELECT AVG(price) AS average_price_per_night
FROM airbnb;

average_price_per_night
0               140.244375

In [8]:
%%sql
SELECT rating
FROM airbnb
GROUP BY rating
ORDER BY rating DESC
LIMIT 10;

rating
0  4.999561
1  4.999229
2  4.998898
3  4.998890
4  4.998753
5  4.997670
6  4.997668
7  4.996448
8  4.996384
9  4.995920

In [9]:
%%sql
SELECT AVG(reviews_per_month) AS average_reviews_per_month
FROM airbnb;

average_reviews_per_month
0                   1.353746

In [10]:
%%sql
SELECT AVG(availability_365) AS average_availability_per_year
FROM airbnb;

average_availability_per_year
0                      114.54616

In [11]:
%%sql
SELECT COUNT(DISTINCT listing_id) AS unique_listings
FROM airbnb;

unique_listings
0             7734

In [12]:
%%sql
SELECT COUNT(DISTINCT host_id) AS unique_hosts
FROM airbnb;

unique_hosts
0          7023

1. **Average Price per Night:**
   - The average price per night across all Airbnb listings in the dataset is approximately $140.24.

2. **Distribution of Ratings:**
   - The distribution of ratings across the listings in the dataset shows the count of listings for each rating value from 0 to 5.

3. **Average Number of Reviews per Month:**
   - The average number of reviews per month across all Airbnb listings in the dataset is approximately 1.35.

4. **Average Availability per Year:**
   - The average availability per year across all Airbnb listings in the dataset is approximately 114.55 days.

5. **Unique Listings and Hosts:**
   - The dataset contains 7,734 unique listings.
   - There are 7,023 unique hosts in the dataset.

These analyses provide insights into the pricing, ratings, review frequency, availability, and uniqueness of listings and hosts in the dataset.